### PR TITLE
4/ Implemented entity conversion

### LINKED ISSUE #191: Convert EnemyNPC to FriendlyNPC once the enemy is beaten
# **Description**
**Task:** Once an EnemyNPC is beaten, change the entity type of the beaten NPC to a FriendlyNPC
**Feature:** Friendly Animals (NPC) [Team 4]

This task involves modifying the game's logic to recognise when an EnemyNPC has been defeated. Upon defeat, the EnemyNPC's entity type will be updated to FriendlyNPC. This change will trigger corresponding behaviors and interactions, such as preventing the FriendlyNPC from attacking the player and potentially introducing new gameplay elements or interactions. This feature will add a dynamic element to the game world by allowing defeated enemies to become potential allies or sources of information.

# **Dependencies**
- Team 5 for Combat Interaction
 
# **Milestones**
List of steps that need to be completed for this task.
- [x] Implement a class for the conversion of the entities (Sep. 2)
- [x] Implement a check to see if an NPC has been defeated (Sep. 5)
- [x] Convert EnemyNPC to FriendlyNPC (Sep. 5)
- [x] Implement testing for the conversion of entities (Sep. 9)
- [x] Write up the documentation for the new class in the wiki (Sep. 10)
 
**Completion Deadline:** Sep. 11
 
# **Documentation**
[Entity Conversion Details](https://github.com/UQcsse3200/2024-studio-2/wiki/Enemy-Converting-to-Friendly-Upon-Defeat)
 
# **Member**
- Shubh Gupta (@shubhgupta2510)

### PR DESCRIPTION
**Pull request by shubhgupta2510**
Closes #191 and #277 and implements the conversion of enemy entities to friendly entities once defeated. Currently, once the enemy is defeated, the game deletes the enemy entity but the EntityConverter class works as expected and should show the enemy as a friendly during Sprint 3.

### CODE REVIEW

File: source/core/src/test/com/csse3200/game/entities/EntityConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
+package com.csse3200.game.entities;
+
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.math.Vector2;
+import com.csse3200.game.ai.tasks.AITaskComponent;
+import com.csse3200.game.components.CombatStatsComponent;
+import com.csse3200.game.components.TouchAttackComponent;
+import com.csse3200.game.components.npc.ChickenAnimationController;
+import com.csse3200.game.components.npc.FriendlyNPCAnimationController;
```
Comment (PHONGNGYUEN1) [inline source/core/src/test/com/csse3200/game/entities/EntityConverterTest.java:9]: Unused file here

**Review by PHONGNGYUEN1 — APPROVED** [review]

**Comment by PHONGNGYUEN1** [comment]
These tests are failing please fix before pushing

**Review by PHONGNGYUEN1 — CHANGES_REQUESTED** [review]
See comm

File: source/core/src/test/com/csse3200/game/entities/EntityConverterTest.java (hunk 1 above)
Comment (Jul-Wong) [reply source/core/src/test/com/csse3200/game/entities/EntityConverterTest.java:9]: Which tests are failing?

Comment (Jul-Wong) [reply source/core/src/test/com/csse3200/game/entities/EntityConverterTest.java:9]: lol nvm i see

**Review by PHONGNGYUEN1 — APPROVED** [review]

**Review by AndreBissell — APPROVED** [review]
looks good

**Review by mhUnt3rS — COMMENTED** [review]
looks good

**Pull request merged**